### PR TITLE
Prepare release 5.4.0

### PR DESCRIPTION
**Pull request by pytestbot**
Created automatically from https://github.com/pytest-dev/pytest/issues/6892#issuecomment-597621121.

Once all builds pass and it has been **approved** by one or more maintainers, the build
can be released by pushing a tag `5.4.0` to this repository.

### CODE REVIEW

File: doc/en/example/simple.rst [hunk 1]
```diff
@@ -619,57 +607,10 @@ We can run this:
...
-        def test_a2(db):
-    >       assert 0, db  # to show value
-    E       AssertionError: <conftest.DB object at 0xdeadbeef>
-    E       assert 0
-
-    a/test_db2.py:2: AssertionError
-    ============= 3 failed, 2 passed, 1 xfailed, 1 error in 0.12s ==============
+    ImportError while loading conftest '$REGENDOC_TMPDIR/conftest.py'.
+    conftest.py:3: in <module>
+        _test_failed_incremental: Dict[str, Dict[Tuple[int, ...], str]] = {}
```
Comment (nicoddemus) [inline doc/en/example/simple.rst]: Ditto

File: doc/en/writing_plugins.rst [hunk 2]
```diff
@@ -442,8 +442,13 @@ additionally it is possible to copy examples for an example folder before runnin
       $REGENDOC_TMPDIR/test_example.py:4: PytestExperimentalApiWarning: testdir.copy_example is an experimental api that may change over time
         testdir.copy_example("test_example.py")
 
+    test_example.py::test_plugin
```
Comment (nicoddemus) [inline doc/en/writing_plugins.rst]: We should change the example to avoid this warning.

File: doc/en/example/parametrize.rst [hunk 3]
```diff
@@ -607,13 +613,13 @@ Then run ``pytest`` with verbose mode and with only the ``basic`` marker:
     platform linux -- Python 3.x.y, pytest-5.x.y, py-1.x.y, pluggy-0.x.y -- $PYTHON_PREFIX/bin/python
     cachedir: $PYTHON_PREFIX/.pytest_cache
     rootdir: $REGENDOC_TMPDIR
-    collecting ... collected 17 items / 14 deselected / 3 selected
+    collecting ... collected 14 items / 11 deselected / 3 selected
```
Comment (RonnyPfannschmidt) [inline doc/en/example/parametrize.rst]: why did this number change?

**Comment by bluetech** [comment]
I didn't encounter any regressions when using pytest from this branch on our test suite (both with and without syntax highlighting).

**Comment by nicoddemus** [comment]
I also did not encounter any regressions in the projects I tested. 👍

**Comment by bluetech** [comment]
But the PR shouldn't target master, it should target a `5.4.x` branch. At least that's what we finally agreed on in the RELEASING.md file. I guess this causes an issue that the new docs and CHANGELOG.md are not reflected in master...

**Comment by nicoddemus** [comment]
> But the PR shouldn't target master, it should target a 5.4.x branch. At least that's what we finally agreed on in the RELEASING.md file.

Oh I was under the impression that we would create `5.4.x` before releasing `5.5.0` (one of the reasons is the changelog).

Either way we can make the release branch target master, and create `5.4.x` afterwards?

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus it accidentially reuses the pr and i see a failure on the execution

**Comment by nicoddemus** [comment]
>  it accidentially reuses the pr and i see a failure on the execution

I think it is good to re-use actually... about the failure I will have to take a look.

**Comment by nicoddemus** [comment]
Unfortunately the failure is again on check-links:

```
/home/runner/work/pytest/pytest/doc/en/projects.rst.rst:43: WARNING: broken link: http://code.google.com/p/py-s3fuse/ (500 Server Error: Internal Server Error for url: http://code.google.com/p/py-s3fuse/)
```

**Comment by The-Compiler** [comment]
So it looks like [this was released](https://mail.python.org/pipermail/pytest-dev/2020-March/004914.html) despite the [issue](https://github.com/pytest-dev/pytest/issues/6892#issuecomment-597622909) I've reported about pytest-rerunfailures (used by >1k repositories according to GitHub)? So far we always gave plugin authors a couple of days to fix issues if we noticed anything before a release...

**Comment by RonnyPfannschmidt** [comment]
Indeed, I missed your comment, let's ensure we use a change request next time, im not sure if i can provide a fix tonight but i believe it is related to a typing fixup we can pass on swiftly

**Comment by The-Compiler** [comment]
Note that the PR didn't even exist when I posted the comment. If we want to do QA testing via PR change requests in the future, whoever is taking care of the release should probably make sure there's enough time for people to notice and test it, which IMHO means >24h.

**Comment by The-Compiler** [comment]
Oh, also I just noticed that https://github.com/pytest-dev/pytest/releases only shows "Preparing release version 5.4.0" instead of a changelog.

**Comment by bluetech** [comment]
@RonnyPfannschmidt , do you intend to merge this PR? Right now the tag is only on this open PR (`5.4.0-release` branch). It will be good to have it in `master` (so repo-browsers can see the CHANGELOG.md) or in a `5.4.x` branch.

**Comment by nicoddemus** [comment]
> which IMHO means >24h.

Good point, let's try to improve that for the next release where we want some more intensive QA. 👍

**Comment by nicoddemus** [comment]
> It will be good to have it in master (so repo-browsers can see the CHANGELOG.md) or in a 5.4.x branch.

I've pushed `5.4.x` from this point so we can backport fixes to it. 👍

**Comment by nicoddemus** [comment]
> Oh, also I just noticed that pytest-dev/pytest/releases only shows "Preparing release version 5.4.0" instead of a changelog.

While investigating this, I found out that the script that publishes the release notes was failing because by the time it executed the release was already created, here's the error code returned by `github3`:

```python
[{'resource': 'Release', 'code': 'already_exists', 'field': 'tag_name'}]
```

Which is weird. Did you create the release manually somehow @RonnyPfannschmidt, by any chance?

Anyway I've updated it with the proper changelog text:

https://github.com/pytest-dev/pytest/releases/tag/5.4.0

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus yes, my misstake

**Comment by nicoddemus** [comment]
Oh OK, no worries! Better this way, it means the script still works. 😅

**Pull request merged**